src/RecursiveApply/RecursiveApply.jl

```diff
@@ -14,26 +14,37 @@ export ⊞, ⊠, ⊟
 
 Recursively apply `fn` to each element of `X`
 """
-rmap(fn, X) = fn(X)
-rmap(fn, X, Y) = fn(X, Y)
-rmap(fn, X::Tuple) = map(x -> rmap(fn, x), X)
-rmap(fn, X::Tuple, Y::Tuple) = map((x, y) -> rmap(fn, x, y), X, Y)
-rmap(fn, X::NamedTuple) = map(x -> rmap(fn, x), X)
-rmap(fn, X::NamedTuple{names}, Y::NamedTuple{names}) where {names} =
-    map((x, y) -> rmap(fn, x, y), X, Y)
+rmap(fn::F, X) where F = fn(X)
+rmap(fn::F, X, Y) where F = fn(X, Y)
+rmap(fn::F, X::Tuple) where F = map(x -> rmap(fn, x), X)
+rmap(fn::F, X::Tuple, Y::Tuple) where F = map((x, y) -> rmap(fn, x, y), X, Y)
+function rmap(fn::F, X::NamedTuple{names}) where {F, names}
+    tup = ntuple(Val(length(X))) do i
+        Base.@_inline_meta
+        fn(X[i])
+    end
```

Comment:
Does this work?
```suggestion
    tup = rmap(fn, Tuple(X))
```

Reply:
This one works (it is used by `roeflux` so I've verified that). The next one (`rmap(fn, X, Y)`) I haven't tested.

Reply:
So yes, this works. Loses the added security of the `Base.@_inline_meta` but that may not matter.

Prefer it? And the same for the `rmap(fn, X, Y)` form below?

Reply:
Yes, it's a bit simpler